pycyphal/transport/commons/high_overhead_transport/_transfer_reassembler.py

```diff
@@ -181,9 +207,23 @@ def process_frame(
             frame_payloads=self._payloads,
             source_node_id=self._source_node_id,
         )
+
+        # determine (possible) error
+        error = None
+        if result is None:
+            assert len(self._payloads) >= 1
+            if len(self._payloads) > 1:
+                error = self.Error.MULTIFRAME_INTEGRITY_ERROR
+            elif len(self._payloads) == 1:
+                error = self.Error.UNIFRAME_INTEGRITY_ERROR
```

Comment:
This distinction appears artificial so I suggest eliminating it by combining `MULTIFRAME_INTEGRITY_ERROR` and `UNIFRAME_INTEGRITY_ERROR` as `INTEGRITY_ERROR`.